docstore/query.go

```diff
@@ -196,20 +197,100 @@ func (q *Query) Delete(ctx context.Context) error {
 	if err := q.validateWrite("delete"); err != nil {
 		return err
 	}
-	return q.coll.driver.RunDeleteQuery(ctx, q.dq)
+	type rdq interface {
+		RunDeleteQuery(context.Context, *driver.Query) error
+	}
+
+	if d, ok := q.coll.driver.(rdq); ok {
+		return d.RunDeleteQuery(ctx, q.dq)
+	}
+
+	return q.runAction(ctx, func(al *ActionList, doc Document) {
+		al.Delete(doc)
+	})
+}
+
+func (q *Query) runAction(ctx context.Context, addAction func(*ActionList, Document)) error {
```

Comment:
`runQueryActions`?

Reply:
runAndRetryActions?

Reply:
Went with a variation on polinasok's. Didn't want to use the word "Query" since this is a method on Query.